v3/src/components/graph/adornments/univariate-measures/univariate-measure-adornment-component.tsx

```diff
@@ -0,0 +1,245 @@
+
+    valueObj.line = selection.append("line")
+      .attr("class", lineClass)
+      .attr("id", lineId)
+      .attr("data-testid", lineId)
+      .attr("x1", x1)
+      .attr("x2", x2)
+      .attr("y1", y1)
+      .attr("y2", y2)
+
+    valueObj.cover = selection.append("line")
+      .attr("class", coverClass)
+      .attr("id", coverId)
+      .attr("data-testid", coverId)
+      .attr("x1", x1)
+      .attr("x2", x2)
+      .attr("y1", y1)
+      .attr("y2", y2)
+
+    if (showLabel) {
+      const labelCoords = adornment.labelCoords
+      const activeUnivariateMeasures = adornmentsStore?.activeUnivariateMeasures
+      const adornmentIndex = activeUnivariateMeasures?.indexOf(model.type) ?? null
+      const topOffset = activeUnivariateMeasures.length > 1 ? adornmentIndex * 20 : 0
+      const left = labelCoords ? labelCoords.x / xCellCount : isVertical.current ? xScale(plotValue) / xCellCount : 0
+      const top = labelCoords ? labelCoords.y : topOffset
+      const labelId = `${measureType}-measure-labels-tip-${containerId}${classFromKey ? `-${classFromKey}` : ""}`
+      const labelClass = clsx("measure-labels-tip", `measure-labels-tip-${measureType}`)
+      labelObj.label = labelSelection.append("div")
+        .text(`${measureType}=${displayValue}`)
+        .attr("id", labelId)
+        .attr("class", labelClass)
+        .attr("data-testid", labelId)
+        .style("left", `${left}px`)
+        .style("top", `${top}px`)
+
+      labelObj.label.call(
+        drag<HTMLDivElement, unknown>()
+          .on("drag", (e) => handleMoveLabel(e, labelId))
+          .on("end", (e) => handleEndMoveLabel(e, labelId))
+      )
+
+      valueObj.cover.on("mouseover", (e) => highlightLabel(labelId, true))
+        .on("mouseout", (e) => highlightLabel(labelId, false))
+
+      labelObj.label.on("mouseover", (e) => highlightCover(coverId, true))
+        .on("mouseout", (e) => highlightCover(coverId, false))
+    } else {
+      const x = isVertical.current ? xScale(plotValue) + 5 : plotWidth - (plotWidth/2)
+      const y = isVertical.current ? 50 : yScale(plotValue) - 5
+      const textId = `${measureType}-tip-${containerId}${classFromKey ? `-${classFromKey}` : ""}`
+      const textClass = clsx("measure-tip", `${measureType}-tip`)
+      valueObj.text = selection.append("text")
+        .text(`${measureType}=${displayValue}`)
+        .attr("id", textId)
+        .attr("class", textClass)
+        .attr("data-testid", textId)
+        .attr("x", x / xCellCount)
+        .attr("y", y / yCellCount)
+    }
+
+  }, [adornmentsStore?.activeUnivariateMeasures, classFromKey, containerId, handleEndMoveLabel,
+      handleMoveLabel, highlightCover, highlightLabel, layout, measureType, model.type, plotHeight,
+      plotWidth, showLabel, xAttrType, xScale, yAttrType, yScale])
+
+  // Add the lines and their associated covers and labels
+  const refreshValues = useCallback(() => {
+    const disposer = autorun(() => {
```

Comment:
I don't think you want the `autorun` here. `refreshValues` should just be a function that runs when called and then any necessary `autorun`s or `reaction`s would be installed in `useEffect`s like `refreshAxisChange` below.